File: seearchingAndSorting/InterpolationSearch.js
let arr = [10, 20, 30, 40, 50];
let val = 60; 
function interpolation(arr , low , high){
    while(low<=high && val>=arr[low] && val<=arr[high] ){
        pos = low + ((val - arr[low]) * (high - low)) / (arr[high] - arr[low]) 

        if(arr[pos]==val){
            return pos;
        }
        else if(arr[pos] < val){
            low = pos+1
        }
        else{
            high = pos-1
        }

    }
     return -1
}


let ans = interpolation(arr, 0, arr.length-1)
if(ans == -1){
    console.log("not found")
}
else{
    console.log("found at index ", + ans);
}

File: seearchingAndSorting/interpolationSearch.cpp
#include<iostream>
using namespace std;

int interpolationSearch(int arr[], int key, int start, int end){
    while(start <= end ){
        int position = start + ((key - arr[start]) * (end - start)) / (arr[end] - arr[start]);
        if(arr[position]==key){
            return position;
        }
        else if(arr[position]<key){
            start = position+1;
        }
        else{
            end = position-1;
        }
    }
    return -1;
}

int main(){
    int arr[] = {20, 40,60, 80, 100};
    int key = 80;
    int n = sizeof(arr)/ sizeof(arr[0]);
    int index = interpolationSearch(arr,key,0,n-1);
    if(index==-1){
        cout<<"not found";
    }
    else{
        cout<<"found at index : "<<index;
    }
}
